sdk/appcenter/src/main/java/com/microsoft/appcenter/channel/DefaultChannel.java

```diff
@@ -756,12 +769,54 @@ synchronized void checkPendingLogs(@NonNull GroupState groupState) {
             return;
         }
         long pendingLogCount = groupState.mPendingLogCount;
-        AppCenterLog.debug(LOG_TAG, "checkPendingLogs(" + groupState.mName + ") pendingLogCount=" + pendingLogCount);
-        if (pendingLogCount >= groupState.mMaxLogsPerBatch) {
+        long batchTimeInterval = groupState.mBatchTimeInterval;
+        AppCenterLog.debug(LOG_TAG, String.format("checkPendingLogs(%s) pendingLogCount=%s batchTimeInterval=%s", groupState.mName, pendingLogCount, batchTimeInterval));
+
+        /* If the interval is custom. */
+        if (batchTimeInterval > MINIMUM_TRANSMISSION_INTERVAL) {
```

Comment:
There was a discussion on slack for the corner cases where transmission interval is custom and we have a lot of logs. We still need to send multiple batches when the timer is due.

When I read this code this will just sent 50 logs a day (if sending every day) and will never be  able to flush queue if we get a lot.

When timer is due we must send as many batches as necessary and this does not seem to be handled.

Reply:
To rephrase: send 200 logs and have transmission once day: we must still send 3 parallel batches, then the 4th immediately once one of them completes.

Reply:
Also must avoid infinite sending if you keep enqueueing while "flushing", just logs enqueued before starting first batch should be sent, so use the timestamp of when timer is due (not started) to query logs is one solution to avoid that. (All logs enqueued after we started flushing will be for next timer).

Reply:
@guperrot 
> send 200 logs and have transmission once day: we must still send 3 parallel batches, then the 4th immediately once one of them completes.

It should work now

Reply:
https://github.com/microsoft/appcenter-sdk-android/pull/1161/files#diff-0ebb2cb4967c7aee91e0b189502daf32R800

Reply:
Yes the code is confusing but it works. Unit tests should reflect this.